### ISSUE TITLE
PayPal Request failt

### ISSUE THREAD

**Issue by maerlynflagg** [nopSolutions/PayPal-SmartButtons-plugin-for-nopcommerce#14]
i followed your documentation to configure the PayPal smart buttons.

in paypal i created the LIVE API Client ID and Secret and entered it in the your plugin configuration. i disabled the sandbox option.

but i got an error:

`Request is not well-formed, syntactically incorrect, or violates schema.`

i tried capture and authorize. 

i'm using...

```
NopCommerce 4.3
Plugin Version 1.13
```

what's going wrong?

**Comment by RomanovM** [comment]
@maerlynflagg Check your logs, there should be more detailed error, something about incorrect data in the order.
Anyway, please post your questions on [our forums](https://www.nopcommerce.com/boards/) (see also the [similar topics](https://www.nopcommerce.com/en/boards/search?searchterms=Request+is+not+well-formed)), we only use GitHub as an issue tracker.

**Comment by maerlynflagg** [comment]
there is no reason to close it. this isn't a question for a forum. this is a bug, if your plugin not working as expected, if all data entered correctly. 

i know your forum, i googled the threads. for example i know this one: https://www.nopcommerce.com/en/boards/topic/87866/configure-paypal-smart-payment-buttons

so it's not helpful , that this bug only fixed for older version, but not for the new version.

**Comment by maerlynflagg** [comment]
backend: alle required configuration fields are set, which is described in your documentation.

the request is formed by the plugin, so this is definitively not a "ask to forum" problem.  i know, for what github is and this is a bug, also an issue.


Log:

```
System.AggregateException: One or more errors occurred. ({"name":"INVALID_REQUEST","message":"Request is not well-formed, syntactically incorrect, or violates schema.","debug_id":"6d1e1fdaf4e7c","details":[{"field":"/payer/address/country_code","value":"","location":"body","issue":"MISSING_REQUIRED_PARAMETER","description":"A required field / parameter is missing."},{"field":"/purchase_units/@reference_id=='42a84329-e151-445a-b38c-ba4a863b9826'/shipping/address/country_code","value":"","location":"body","issue":"MISSING_REQUIRED_PARAMETER","description":"A required field / parameter is missing."}],"links":[{"href":"https://developer.paypal.com/docs/api/orders/v2/#error-MISSING_REQUIRED_PARAMETER","rel":"information_link","encType":"application/json"}]})
 ---> PayPalHttp.HttpException: {"name":"INVALID_REQUEST","message":"Request is not well-formed, syntactically incorrect, or violates schema.","debug_id":"6d1e1fdaf4e7c","details":[{"field":"/payer/address/country_code","value":"","location":"body","issue":"MISSING_REQUIRED_PARAMETER","description":"A required field / parameter is missing."},{"field":"/purchase_units/@reference_id=='42a84329-e151-445a-b38c-ba4a863b9826'/shipping/address/country_code","value":"","location":"body","issue":"MISSING_REQUIRED_PARAMETER","description":"A required field / parameter is missing."}],"links":[{"href":"https://developer.paypal.com/docs/api/orders/v2/#error-MISSING_REQUIRED_PARAMETER","rel":"information_link","encType":"application/json"}]}
   at PayPalHttp.HttpClient.Execute[T](T req)
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore(Boolean waitCompletionNotification)
   at Nop.Plugin.Payments.PayPalSmartPaymentButtons.Services.ServiceManager.HandleCheckoutRequest[TRequest,TResult](PayPalSmartPaymentButtonsSettings settings, TRequest request)
   at Nop.Plugin.Payments.PayPalSmartPaymentButtons.Services.ServiceManager.<>c__DisplayClass22_0.<CreateOrder>b__0()
   at Nop.Plugin.Payments.PayPalSmartPaymentButtons.Services.ServiceManager.HandleFunction[TResult](PayPalSmartPaymentButtonsSettings settings, Func`1 function)
```

**Comment by RomanovM** [comment]
@maerlynflagg The log error you provided is self-describing, it's indicated that the `country_code` is missing in the address when placing an order. This could happen if [CountryEnabled](https://admin-demo.nopcommerce.com/Admin/Setting/AllSettings?settingName=addresssettings.countryenabled) setting is false or country details are incorrectly filled in the admin area. 
The request is formed by the plugin, but the order details are provided by the customer. In this case, the plugin only indicates errors in the order and this is definitely not a bug.

As for the above topic with a similar error, it has been fixed for both versions [4.20](https://github.com/nopSolutions/PayPal-SmartButtons-plugin-for-nopcommerce/blob/nopCommerce-4.20/Nop.Plugin.Payments.PayPalSmartPaymentButtons/Services/ServiceManager.cs#L303) and [4.30](https://github.com/nopSolutions/nopCommerce/blob/master/src/Plugins/Nop.Plugin.Payments.PayPalSmartPaymentButtons/Services/ServiceManager.cs#L312). And it has nothing to do with your example, it was about phone number format.

In any case, it's better to post bug-reports, suggestions and other questions on the forums, our active and helpful community will always help, if there is actually a bug we'll create an issue on GitHub ourselves and take it to work. This is the preferred way.
